test/multiprocessing_test_case.py

```diff
@@ -0,0 +1,209 @@
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import logging
+import multiprocessing
+import sys
+import tempfile
+import time
+import unittest
+from collections import namedtuple
+from functools import wraps
+
+import torch
+import torch.distributed as c10d
+from common_utils import TestCase, NO_MULTIPROCESSING_SPAWN
+
+
+TestSkip = namedtuple("TestSkip", "exit_code, message")
+
+
+TEST_SKIPS = {
+    "multi-gpu": TestSkip(75, "Need at least 2 CUDA devices"),
+    "nccl": TestSkip(76, "c10d not compiled with NCCL support"),
+    "known_issues": TestSkip(77, "Test skipped due to known issues"),
+}
+
+
+def skip_if_not_multigpu(func):
+    """Multi-GPU tests requires at least 2 GPUS. Skip if this is not met."""
+
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        if torch.cuda.is_available() and torch.cuda.device_count() >= 2:
+            return func(*args, **kwargs)
+        sys.exit(TEST_SKIPS["multi-gpu"].exit_code)
+
+    return wrapper
+
+
+def skip_if_lt_x_gpu(x):
+    def decorator(func):
+        @wraps(func)
+        def wrapper(*args, **kwargs):
+            if torch.cuda.is_available() and torch.cuda.device_count() >= x:
+                return func(*args, **kwargs)
+            sys.exit(TEST_SKIPS["multi-gpu"].exit_code)
+
+        return wrapper
+
+    return decorator
+
+
+def skip_for_known_issues(func):
+    """Skips a test due to known issues (for c10d)."""
+
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        sys.exit(TEST_SKIPS["known_issues"].exit_code)
+
+    return wrapper
+
+
+def requires_gloo():
+    return unittest.skipUnless(
+        c10d.is_gloo_available(), "c10d was not compiled with the Gloo backend"
+    )
+
+
+def requires_nccl():
+    return unittest.skipUnless(
+        c10d.is_nccl_available(), "c10d was not compiled with the NCCL backend"
+    )
+
+
+def requires_mpi():
+    return unittest.skipUnless(
+        c10d.is_mpi_available(), "c10d was not compiled with the MPI backend"
+    )
+
+
+TIMEOUT_DEFAULT = 100
+TIMEOUT_OVERRIDE = {}
+
+
+def get_timeout(test_id):
+    return TIMEOUT_OVERRIDE.get(test_id.split(".")[-1], TIMEOUT_DEFAULT)
+
+
+class MultiProcessTestCase(TestCase):
```

Comment:
Let's add some comments here to explain the difference between this `MultiProcessTestCase` and the one in `common_distributed.py`. It might be helpful to give them different names.